qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Classical Quantum Real Time Evolution."""
+from typing import Tuple, List, Optional
+import numpy as np
+import scipy.sparse as sp
+from qiskit import QuantumCircuit
+from qiskit.opflow import StateFn
+from qiskit.quantum_info.states import Statevector
+
+
+from ..evolution_problem import EvolutionProblem
+from ..evolution_result import EvolutionResult
+from ..imaginary_evolver import ImaginaryEvolver
+from .scipy_evolver import SciPyEvolver
+
+
+class SciPyImaginaryEvolver(ImaginaryEvolver, SciPyEvolver):
+    r"""Classical Evolver for imaginary time evolution.
+
+    Evolves an initial state :math:`|\Psi\rangle` for an imaginary time :math:`\tau = it`
+    under a Hamiltonian  :math:`H`, as provided in the ``evolution_problem``.
+
+    For each timestep we use a taylor expansion of :math:`\exp(- \Delta \tau H)` to evolve the system.
+
+    """
+
+    def __init__(
+        self,
+        timesteps: Optional[int] = 100,
+    ):
+        """
+        Args:
+            timesteps: The number of timesteps in the simulation.
+
+        Raises:
+            ValueError: If `timesteps` is not a positive integer.
+        """
+        if isinstance(timesteps, int) and timesteps < 1:
+            raise ValueError("`timesteps` must be a positive integer.")
+
+        self.timesteps = timesteps
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        r"""Perform imaginary time evolution :math:`\exp(- \tau H)|\Psi\rangle`.
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            Evolution result which includes an evolved quantum state.
+
+        Raises:
+            ValueError: If a `NaN` is encountered in the final state of the system.
+            RuntimeError: If the Hamiltonian in the evolution problem is time dependent.
+            RuntimeError: If the state norm goes to 0 during time evolution
+        """
+
+        if evolution_problem.t_param is not None:
+            raise ValueError("Time dependent Hamiltonians are not currently supported.")
```

Comment:
Maybe this could be a NotImplementedError? Also the message might be improved to say something like `Time dependent Hamiltonians in the evolution problem are not supported.` to make it clear that the issue stems from the contents of problem. Also I would drop the 'currently' - perhaps we can support this going forwards...

Reply:
Well, I mean it could be supported. It has nothing to do with the problem, but rather the method that we are choosing to integrate the time evolution. In the future it could very well be implemented.

Reply:
Yes, I get that it could be... In the past wording like this would be avoided as it can be read as an expectation that it will be supported when one includes qualifiers like currently. Its not supported with this code whether you include currently or not. Even without currently in the message we could still choose to support it at some point if/when that's possible.